daemon/graphdriver/overlay2/overlay.go

```diff
@@ -174,6 +175,14 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		}
 	}
 
+	// If kernel is older than 4.8, check whether opaque copy up bug has been patched
+	if kernel.CompareKernelVersion(*v, kernel.VersionInfo{Kernel: 4, Major: 8, Minor: 0}) < 0 {
```

Comment:
We can keep that check if you like it, although kernel versions are abused all the time and it is hard to predict who will decide to release something with a wrong version. With lazy loading, the additional time for full check should be very small.